pkg/controllers/cloud/controller_sync_status.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloud
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"antrea.io/nephe/pkg/logging"
+	"github.com/cenkalti/backoff/v4"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var (
+	ctrlSyncStatus *controllerSyncStatus
+)
+
+type controllerSyncStatus struct {
+	mutex   sync.Mutex
+	syncMap map[controllerType]bool
+	// number of values of the type controllerType
+	numElements int
+	log         func() logging.Logger
+}
+
+func init() {
+	ctrlSyncStatus = &controllerSyncStatus{
+		syncMap:     make(map[controllerType]bool),
+		numElements: int(ControllerTypeVM),
+		log: func() logging.Logger {
+			return logging.GetLogger("controllers-sync-status")
+		},
+	}
+
+	// initialize the map to default values
+	for i := 0; i < ctrlSyncStatus.numElements; i++ {
+		ctrlSyncStatus.syncMap[controllerType(i)] = false
+	}
+}
+
+func GetControllerSyncStatusInstance() *controllerSyncStatus {
+	return ctrlSyncStatus
+}
+
+// SetControllerSyncStatus sets the sync status of the controller.
+func (c *controllerSyncStatus) SetControllerSyncStatus(controller controllerType) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
```

Comment:
lets use RWlock?

Reply:
Don't have any strong inclination against a RWLock. Will change it to RWLock.